### PR TITLE
fix: Do not override short name with relative (longer) reference

### PR DESCRIPTION
**Pull request by Wirone**
Fixes 2nd problem from [this comment](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/7744#issuecomment-1893449549). Extracted from #7719.

### CODE REVIEW

**Comment by mvorisek** [comment]
there are more issues, it should be already fixed in https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7719 in a better way

**Comment by Wirone** [comment]
@mvorisek thank you for the information that this problem is also fixed in #7719 - I guess that's the side effect of putting many things in single PR, which is harder to review 🙂. If anyone can review your PR today and approve it _today_, so we can release a patch with it, then I'm OK with it. But we also can merge this one as a low hanging fruit, and then replace it with your implementation. The most important part is us to be able to deliver patch today, and I currently don't have time and focus to review bigger changes.

**Comment by mvorisek** [comment]
My fix has much less lines than this discussion. If you want to go this way, replace this PR by cherrypicking https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/commit/94a29364ce6200f2dc0b6a7b6270269832fe68a3 full commit incl. tests.

**Comment by Wirone** [comment]
> If you want to go this way, replace this PR by full commit incl. tests.

Done.

> Also this PR does not fix `B|null` phpdoc so alone it is more or less useless as a full release fix

Yes, because this PR addresses @soyuka's issue only, as a bug fix after 3.47 that had real-world impact. Even though I often process your PRs that fix multiple issues at once, I believe that multiple smaller PRs are better from reviewer's point of view - I can do it while walking with dog for example 😉. Most probably this fix would land in 3.47 if it was in standalone PR before, because it's a no-brainer, while the other parts of #7719 require more attention 🤷‍♂️.

**Review by soyuka — APPROVED** [review]
Thanks this fixes my issue.

File: tests/Fixer/Import/FullyQualifiedStrictTypesFixerTest.php [hunk 1]
```diff
@@ -845,6 +845,29 @@ class Cl {}
             ['import_symbols' => true],
         ];
 
+        yield 'shorten relative reference to already imported, direct short name' => [
```
Comment (mvorisek) [inline tests/Fixer/Import/FullyQualifiedStrictTypesFixerTest.php:848]: this test is not needed /w https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/commit/94a29364ce6200f2dc0b6a7b6270269832fe68a3

Comment (mvorisek) [reply tests/Fixer/Import/FullyQualifiedStrictTypesFixerTest.php:848]: The impl. basically enforce one format, but still a test case is good - 👍

**Pull request merged**